### PR TITLE
DA1469x watchdog driver

### PR DESCRIPTION
**Pull request by blauret**
This PR is adding the watchdog driver for the Renesas DA1469x microcontroller family

### CODE REVIEW

**Comment by nordicjm** [comment]
Please see https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines - all commits need to be properly labelled as per the guidelines

File: drivers/watchdog/wdt_smartbond.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	if (wdog_smartbond_dev_data.callback) {
+		wdog_smartbond_dev_data.callback(wdog_smartbond_dev_data.wdog_device, 0);
+	}
+}
+#endif /* CONFIG_RUNTIME_NMI */
+
+static int wdg_smartbond_install_timeout(const struct device *dev,
+					 const struct wdt_timeout_cfg *config)
+{
+	struct wdog_smartbond_data *data = WDOG_SMARTBOND_DATA(dev);
```
Comment (martinjaeger) [inline drivers/watchdog/wdt_smartbond.c:68]:
```suggestion
	struct wdog_smartbond_data *data = (struct wdog_smartbond_data *)dev->data;
```
Using a custom macro to get the data pointer is not recommended anymore (even though still used in several drivers).

File: drivers/watchdog/wdt_smartbond.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		/* wait */
+	}
+	SYS_WDOG->WATCHDOG_REG = reload_val;
+
+	return 0;
+}
+
+static int wdg_smartbond_feed(const struct device *dev, int channel_id)
+{
+	struct wdog_smartbond_data *data = WDOG_SMARTBOND_DATA(dev);
```
Comment (martinjaeger) [inline drivers/watchdog/wdt_smartbond.c:108]: See above.

**Review by martinjaeger — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Only got a very minor comment. Otherwise looks very good.

Please note that we are in feature freeze at the moment and this PR will only get merged after v3.3 has been released (planned for 17 Feb 2023). We can still get everything ready in the meantime, though.

**Comment by blauret** [comment]
Thanks for the review. 

> Please note that we are in feature freeze at the moment and this PR will only get merged after v3.3 has been released (planned for 17 Feb 2023). We can still get everything ready in the meantime, though.

Thanks for letting me know.

**Review by martinjaeger — DISMISSED** [review]

**Comment by ydamigos** [comment]
Hi Ben
just with two lines:
```
diff --git a/boards/arm/da1469x_dk_pro/da1469x_dk_pro.dts b/boards/arm/da1469x_dk_pro/da1469x_dk_pro.dts
index e075a04f23..95154f5249 100644
--- a/boards/arm/da1469x_dk_pro/da1469x_dk_pro.dts
+++ b/boards/arm/da1469x_dk_pro/da1469x_dk_pro.dts
@@ -59,6 +59,7 @@

        aliases {
                led0 = &red_led;
+               watchdog0 = &wdog;
        };

        sysclk: system-clock {
diff --git a/boards/arm/da1469x_dk_pro/da1469x_dk_pro.yaml b/boards/arm/da1469x_dk_pro/da1469x_dk_pro.yaml
index 04935f5193..eebf1a7676 100644
--- a/boards/arm/da1469x_dk_pro/da1469x_dk_pro.yaml
+++ b/boards/arm/da1469x_dk_pro/da1469x_dk_pro.yaml
@@ -10,3 +10,4 @@ toolchain:
 supported:
   - arduino_gpio
   - gpio
+  - watchdog
```

you could allow twister to run watchdog_basic_api test cases:

```
./scripts/twister --device-testing --device-serial /dev/ttyUSB0 --device-serial-baud 115200 -p da1469x_dk_pro -T samples/drivers/watchdog -v -v -v

./scripts/twister --device-testing --device-serial /dev/ttyUSB0 --device-serial-baud 115200 -p da1469x_dk_pro -T tests/drivers/watchdog -v -v -v
```

**Comment by blauret** [comment]
> you could allow twister to run watchdog sample and basic test cases:
> 
> ```
> ./scripts/twister --device-testing --device-serial /dev/ttyUSB0 --device-serial-baud 115200 -p da1469x_dk_pro -T samples/drivers/watchdog -v -v -v
> 
> ./scripts/twister --device-testing --device-serial /dev/ttyUSB0 --device-serial-baud 115200 -p da1469x_dk_pro -T tests/drivers/watchdog -v -v -v
> ```

Sorry, it took a while to test it's working fine. but it's done now. PR is updated. Thanks for the suggestion.

**Review by ydamigos — DISMISSED** [review]

**Review by martinjaeger — DISMISSED** [review]

**Comment by carlescufi** [comment]
@blauret please rebase
@aaillet could you please review this?

**Review by ydamigos — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**